backend/inspirehep/records/api/mixins.py

```diff
@@ -13,7 +14,94 @@
+                    try:
+                        table_entries_buffer.append(
+                            AuthorsRecords(
+                                author_id=id["value"],
+                                id_type=id["schema"],
+                                record_id=self.id,
+                            )
+                        )
+                    except KeyError:
+                        LOGGER.exception(
+                            "id entry is missing keys",
+                            recid=self.get("control_number"),
+                            uuid=str(self.id),
+                        )
+        return table_entries_buffer
+
+    def generate_entries_for_conferences_in_authors_records_table(self):
+        """Generates AuthorsRecords objects table based on record data for collaborations"""
+        collaborations_field = "collaborations.value"
+        table_entries_buffer = []
+        for collaboration in self.get_value(collaborations_field, []):
+            table_entries_buffer.append(
+                AuthorsRecords(
+                    author_id=collaboration, id_type="collaboration", record_id=self.id
+                )
+            )
+        return table_entries_buffer
+
+    def update_authors_records_table(self):
+        """Puts all authors ids and collaborations in authors_records table"""
+        deleted_count = self.delete_authors_records_table_entries()
+
+        # check if authors and conferencesshould be added to the table
+        if (
+            self.get("deleted", False)
+            or self.pid_type not in ["lit"]
+            or "Literature" not in self["_collections"]
+        ):
+            LOGGER.info(
+                "Not updating record entries in authors_records table. Record's is not literature.",
+                recid=self.get("control_number"),
+                uuid=str(self.id),
+            )
+            return
+        # add entries from authors and collaborations
+        table_entries_buffer = (
+            self.generate_entries_for_authors_in_authors_records_table()
+        )
+        table_entries_buffer.extend(
+            self.generate_entries_for_conferences_in_authors_records_table()
+        )
+
+        db.session.bulk_save_objects(table_entries_buffer)
+        LOGGER.info(
+            "authors_record table updated for record",
+            recid=self.get("control_number"),
+            uuid=str(self.id),
+            added_rows=len(table_entries_buffer),
+            deleted_rows=deleted_count,
+        )
+
+    def delete_authors_records_table_entries(self):
+        """Clean entries for this record"""
+        return AuthorsRecords.query.filter_by(record_id=self.id).delete()
+
+    def delete(self):
+        self.delete_authors_records_table_entries()
+        super().delete()
+
+    def hard_delete(self):
+        self.delete_authors_records_table_entries()
+        super().hard_delete()
+
+    def update(self, *args, **kwargs):
```

Comment:
You should also add ```disable_relations_update``` and use it here

Reply:
Then add updating entries to this table to the migration ```update_relations``` step.

Reply:
Btw, I still don't get how this would work with Data as we seem to make it Literature specific everywhere because we are checking it is literature at every step :/ but I guess we'll worry about that later

Reply:
We don't want to disable it as this should happen on first step of migration not on second. We want all of this data in the table before first citation calculation.

Reply:
Oh, right. Not the citation calculation but we need it when populating the CitationTable for the self-citation flag.